File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.py
import math

def solution(n):
    answer = ''
    
    n = int(n)
    result = ''
    
    while n > 0:
        n -= 1
        if n%3 == 0:
            num = '1'
        elif n%3 == 1:
            num = '2'
        elif n%3 == 2:
            num = '4'
        else:
            pass
        result += num
        n //= 3
    
    answer = result[::-1]
        
    return answer

File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.java
import java.util.*;

class Solution {
    public String solution(int n) {
        StringBuilder answer = new StringBuilder();
        
         while (n > 0) {
            n -= 1;

            if (n % 3 == 0) answer.append("1");
            else if (n % 3 == 1) answer.append("2");
            else answer.append("4");

            n /= 3;
        }
        
        return answer.reverse().toString();
    }
}